### PR TITLE
delete a comment by comment_id

### PR DESCRIPTION
**Pull request by davegreensmith**

### CODE REVIEW

File: models/models.comments.js [hunk 1]
```diff
@@ -36,3 +36,12 @@ exports.checkUserExists = (username) => {
     } else return true;
   });
 };
+
+exports.removeCommentByCommentId = (comment_id) => {
+  return Promise.all([db.query('SELECT * FROM comments WHERE comment_id=$1', [comment_id]), comment_id]).then(([{ rows }, comment_id]) => {
+    if (rows.length === 0) {
+      return Promise.reject({ status: 404, msg: 'Comment not found' });
+    }
+    return db.query('DELETE FROM comments WHERE comment_id = $1', [comment_id]);
```
Comment (CatHoang) [inline models/models.comments.js:45]: I would suggest doing your delete first and then checking if that comment exist after to make sure it has been deleted

**Review by CatHoang — COMMENTED** [review]
Hi Dave, just a small suggestion in your model :)

File: models/models.comments.js [hunk 2]
```diff
@@ -36,3 +36,13 @@ exports.checkUserExists = (username) => {
     } else return true;
   });
 };
+
+exports.removeCommentByCommentId = (comment_id) => {
+  return Promise.all([db.query('DELETE FROM comments WHERE comment_id = $1 RETURNING *', [comment_id]), comment_id]).then(([{ rows }, comment_id]) => {
+    return Promise.all([db.query('SELECT * FROM comments WHERE comment_id=$1', [comment_id]), rows]).then(([{ rows }, deleteResult]) => {
```
Comment (emilyb93) [inline models/models.comments.js:42]: We dont actually need this line. RETURNING * in an SQL query will return all the rows that have been affected/deleted. If the rows have a length of 0, then we can assume that the comment didnt exist in the first place, where as if we get a length of 1, then it did exist and has now been deleted.

**Review by emilyb93 — CHANGES_REQUESTED** [review]
^

**Comment by two-triangles** [comment]
looks good :) Happy to merge :)

**Pull request merged**